app/bundles/LeadBundle/Model/LeadModel.php

```diff
@@ -2173,4 +2173,34 @@ public function addToCompany(Lead $lead, $company)
 
         return false;
     }
+
+    /**
+     * @param Lead $lead
+     * @param $score
+     *
+     * @return bool
+     */
+    public function scoreContactsCompany(Lead $lead, $score)
+    {
+        $success          = false;
+        $entities         = [];
+        $contactCompanies = $this->companyModel->getCompanyLeadRepository()->getCompaniesByLeadId($lead->getId());
```

Comment:
Can we fetch a list of entities once here rather than querying for a list of IDs then querying/hydrating company entities one by one?